nntrainer/layers/layer_node.h

```diff
@@ -252,6 +253,13 @@ class LayerNode final : public ml::train::Layer, public GraphNode {
    */
   bool getFlatten() const;
 
+  /**
+   * @brief Get the Shared From property of the layer node
+   *
+   * @return std::string node name where the weights are burrowed
```

Comment:
shared from will only share the weights in this scheme, but we need to share the hidden state as well for RNNs.

Reply:
We have discussed offline that RNN might need to go through some tweaks but in general it will be fine :)